### PR TITLE
Generalize `zea.ops.PatchedGrid` to `zea.ops.Map`

### PR DESCRIPTION
**Pull request by wesselvannierop**
Introducing `zea.ops.Map` -- this is a type of Pipeline, that maps its operations over specified input arguments. This generalizes `PatchedGrid`, which I have kept that for backwards compatibility, but it is implement as a subclass of `zea.ops.Map`.

You can now specify any keys (together with `in_axes` and `out_axes`, like in jax.vmap) to map over, not just `flatgrid` and `flat_pfield`!

To do this I had to make the reshaping of the grid a separate operation `zea.ops.ReshapeGrid`. So to run your old Pipelines, you now have to append that after `PatchedGrid`! This is one of the steps towards #143.

Additionally, this PR will `log.debug` whenever you give the pipeline keys that it will not use, useful if you think some operation is using them but is actually not.

## Summary by CodeRabbit

* **New Features**
  * Flexible batching operator with explicit controls for input/output axes, chunking and batch sizing.
  * New pipeline reshape step available in reconstruction/beamforming flows.
  * One-time warnings for unexpected pipeline inputs.

* **Improvements**
  * Simplified beamforming workflows by removing implicit reshaping and exposing batching controls.
  * Pipeline configs and notebook outputs now surface batching and logging details for easier tuning.

* **Documentation**
  * Corrected grid dimension naming in docs.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: zea/ops.py [hunk 1]
```diff
@@ -1256,31 +1278,72 @@ def pipeline_to_yaml(pipeline: Pipeline, file_path: str) -> None:
...
+        operations: List[Operation],
+        argnames: list,
+        in_axes=0,
+        out_axes=0,
+        chunks=None,
+        batch_size=None,
+        **kwargs,
+    ):
+        super().__init__(operations, **kwargs)
+        self.argnames = argnames
```
Comment (benluijten) [inline zea/ops.py]: For my understanding:
Does Map chunk all tensors/keys that are given by argnames? And if so, does this mean that for all tensors/keys the same axes are used? 

I'm thinking of the scenerio where you want to map over e.g. transmits, which is typically axis index 1 ([n_batch, n_tx, ...]. And apodizations which might be shaped [n_tx, n_el], aka the transmits axis is index 0.

**Review by tristan-deep — CHANGES_REQUESTED** [review]
see comment above, requires some clarification on usage in the docs.

File: zea/ops.py (hunk 1 above)
Comment (wesselvannierop) [reply zea/ops.py]: This you can use `in_axes` for, like in `jax.vmap`

**Comment by wesselvannierop** [comment]
@benluijten @tristan-deep I've added a snippet and improved the documentation. Let me know if you think this is sufficient!
